app/controllers/agent/ManageAgentController.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.agent
+
+import actions.AuthenticatedAction
+import binders.propertylinks.GetPropertyLinksParameters
+import config.ApplicationConfig
+import controllers.{PaginationParams, PropertyLinkingController}
+import form.EnumMapping
+import javax.inject.{Inject, Named}
+import models.{RepresentationApproved, RepresentationPending}
+import models.propertyrepresentation.{AgentList, AgentOrganisation, AgentSummary, ManageAgentOptions, ManageAgentRequest}
+import models.searchApi.AgentPropertiesFilter.Both
+import models.searchApi.{AgentPropertiesParameters, OwnerAuthResult}
+import play.api.Logger
+import play.api.data.{Form, Forms}
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Request}
+import services.AgentRelationshipService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.propertylinking.errorhandler.CustomErrorHandler
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class ManageAgentController @Inject()(
+      val errorHandler: CustomErrorHandler,
+      authenticated: AuthenticatedAction,
+      agentRelationshipService: AgentRelationshipService,
+      manageAgentPage: views.html.propertyrepresentation.manage.manageAgent,
+      manageAgentZeroPropertyLinks: views.html.propertyrepresentation.manage.manageAgentZeroPropertyLinks,
+      manageAgentOnePropertyLinkWithAgentAssigned: views.html.propertyrepresentation.manage.manageAgentOnePropertyLinkWithAgentAssigned)(
+      implicit override val messagesApi: MessagesApi,
+      override val controllerComponents: MessagesControllerComponents,
+      executionContext: ExecutionContext,
+      val config: ApplicationConfig
+) extends PropertyLinkingController {
+
+  val logger = Logger(this.getClass.getName)
+
+  def manageAgent(agentCode: Option[Long]): Action[AnyContent] = authenticated.async { implicit request =>
```

Comment:
why is agent code optional?

Reply:
it is optional because we won't necessarily have the agentCode at the start. If you look the table in the story (https://jira.tools.tax.service.gov.uk/browse/VTCCA-3204). The scenarios where we are meant to skip the "new agents list" page.

Reply:
the scenario where the user only has one property?